plugins/DNS/DNSResolver.py

```diff
@@ -0,0 +1,103 @@
+from Config import config
+
+import logging
+import time
+import json
+import re
+import os
+
+log = logging.getLogger('DNSPlugin')
+
+class DNSResolver:
+    loaded = False
+    cache = {}
+
+    def __init__(self, site_manager, nameservers, configure):
+        self.site_manager = site_manager
+        self.nameservers = nameservers
+        self.configure = configure
+
+    def load(self):
+        if not self.loaded:
+            self.loadModule()
+            self.loadCache()
+
+            self.resolver = dns.resolver.Resolver(configure=self.configure)
+
+            if not self.configure:
+                self.resolver.nameservers = self.nameservers
+
+            self.loaded = True
+
+    def loadModule(self):
+        import dns.resolver
+        import dnslink
+
+        if config.tor == 'always':
+            response = type('lamdbaobject', (object,), {})()
+            response.flags = dns.flags.TC
```

Comment:
Personally, I don't like the use of type dynamic construction here. I think it might defeat some optimizations, though they probably won't matter because this code is only executed once. Still, `type()` call looks like magic here because we can just use a class (*and* we don't even define the attribute via `type` call). My proposal:

```python
class Response:
    flags = dns.flags.TC
query = lambda *x, **y: Response()
dns.query.udp = query
```

Reply:
Fixed!